crates/uv/tests/pip_compile.rs

```diff
@@ -4563,16 +4563,15 @@ fn invalid_metadata_requires_python() -> Result<()> {
             .arg("--find-links")
             .arg(context.workspace_root.join("scripts").join("links")), @r###"
     success: false
-    exit_code: 2
+    exit_code: 1
     ----- stdout -----
 
     ----- stderr -----
-    error: Failed to download: validation==2.0.0
-      Caused by: Couldn't parse metadata of validation-2.0.0-py3-none-any.whl from validation==2.0.0
-      Caused by: Failed to parse version: Unexpected end of version specifier, expected operator:
-    12
-    ^^
-
+      × No solution found when resolving dependencies:
+      ╰─▶ Because validation==2.0.0 is unusable because its dependencies are
+          unusable because the package metadata could not be parsed and you
```

Comment:
@zanieb - Is there any way for me to improve this (the triple "because") without larger refactors?

Reply:
Can I make `Dependencies::Unavailable` accept an enum or something?

Reply:
Maybe you could switch it from a string to an enum but uhh I'm not really sure. This is the idea behind a generic incompatibility metadata type in PubGrub.

Reply:
If I _can_ do it, do you mind if I do it?

Reply:
No I don't mind it seems like a step in the right direction

Reply:
I'll add separately.